outport/process/transactionsfee/transactionsFeeProcessor.go

```diff
@@ -157,8 +165,24 @@ func (tep *transactionsFeeProcessor) prepareTxWithResultsBasedOnLogs(
 		return
 	}
 
+	isESDTTransfer := res.Operation == core.BuiltInFunctionESDTTransfer
+	if isESDTTransfer && !hasRefund {
+		gasLimit := tep.txFeeCalculator.ComputeGasLimit(tx)
+		gasUsed := gasLimit + tep.builtInCost.GetESDTTransferBuiltInCost()
+		fee := tep.txFeeCalculator.ComputeTxFeeBasedOnGasUsed(tx, gasUsed)
+
+		txWithResults.GetFeeInfo().SetGasUsed(gasUsed)
+		txWithResults.GetFeeInfo().SetFee(fee)
+	}
+
 	for _, event := range txWithResults.log.GetLogEvents() {
-		if core.WriteLogIdentifier == string(event.GetIdentifier()) && !hasRefund {
+		isWriteLog := core.WriteLogIdentifier == string(event.GetIdentifier())
+		tooMuchGas := false
+		if len(event.GetTopics()) > 1 {
+			tooMuchGas = strings.Contains(string(event.GetTopics()[1]), smartContract.TooMuchGasProvidedMessage)
+		}
+
+		if (isWriteLog && !hasRefund && !isESDTTransfer) || (isWriteLog && tooMuchGas) {
```

Comment:
can you extract this into 1-2 variables?

Reply:
done